README.md

```diff
@@ -1 +1,105 @@
+- high level language for distributed applications
+- custom Wasm runtime
+- blockchain or other global consensus mechanism
+
+IPVM's focus is on execution, networking, data flow, and trust. It is important to make the internals debuggable and friendly where possible, but only when it doesn't conflict with the core aim at this layer: efficiency and abstracting away distributed systems concerns.
+
+IPVM's primary focus is on machines and efficiency. Human-friendliness is important, but many human-oriented features should be pushed into higher layers. These may include high-level APIs, language integration, etc.
+
+# Design Principles
+
+> You can prove anything you want by coldly logical reason — if you pick the proper postulates.
+> 
+> — Asimov, "I, Robot"
+
+1. Proliferate!
+2. Open world / extension without prenegotiation
+3. Participatory: anyone can consume, serve, or both
+4. Learn into our strengths: IPVM is not k8s, so don't try to be
+5. Common things should be simple, and complex things should be possible
+6. Handle low-level details (networking, failure, trust) so that others can focus on business logic
+7. Build in layers with clear audiences (machine-to-machine, end-user, etc)
+
+When in doubt, optimize for proliferation. Get implementations into many hands, find services that already tie-in at the RPC and trust layers ([UCAN]), and avoid having the core IPVM team become a bottleneck for extending the network. [Richard Gabriel] described Unix and C as being like (helpful) computer viruses due to how they spread and integrate with other systems.
+
+IPVM simplifies distributed dataflow. It's tempting to think of it as moving unconstrained centralized code to a logically centralized executor, but IPVM actually lowers the barrier of entry to distributed code. A distributed context is fundamentally different from writing code for a single machine[^LamportsProblem].
+
+[Homestar] is the reference implementation of the IPVM protocol. Homestar is intended to be largely self-contained, and to run on as many systems as possible.
+
+[^LamportsProblem]: As Leslie Lamport says, "a distributed system is one in which the failure of a machine you have never heard of can cause your own machine to become unusable".
+
+## Easy for Who?
+
+> Simplicity is a prerequisite for reliability
+> 
+> — Dijkstra, [How do we tell truths that might hurt?]
+
+> Every application has an inherent amount of complexity that cannot be removed or hidden. Instead, it must be dealt with, either in product development or in user interaction. 
+>
+> — [Tesler's Law]
+
+The word "simple" is often used to mean "to make a common use case easy". This is goal directed: "simple for who or what?" IPVM aims to make distributed computing easy by abstracting away the common challenges of that context. In the same way that TCP/IP handles machine-to-machine networking concerns in an application agnostic way, IPVM doesn't attempt to enforce what can be run on it.
```

Comment:
Do feel like we should make some hit on Hickey's https://www.infoq.com/presentations/Simple-Made-Easy/, which distills these points well (whatever thoughts on clojure may be had).

Reply:
I actually don't like Simple Made Easy for this point. I'm trying to get at simplex versus complex has to be _relative to some domain or goal._ There is no _objective_ simplex, you have to define what you're acting on first, and the choice of domain implies what gets optimised for.

Reply:
I gotcha. Yeah, I think adding a little bit of the clarity you've mentioned here would help tighten this point.